File: samples/dotnet-isolated-azurefunction/OutputBinding/CreateNewOrder.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
// ------------------------------------------------------------

namespace dotnet_isolated_azurefunction
{
    using System.Text.Json;
    using Microsoft.Azure.Functions.Worker;
    using Microsoft.Azure.Functions.Worker.Extensions.Dapr;
    using Microsoft.Extensions.Logging;

    public static class CreateNewOrder
    {
        /// <summary>
        /// Example to use Dapr Service Invocation Trigger and Dapr State Output binding to persist a new state into statestore
        /// </summary>
        [Function("CreateNewOrder")]
        [DaprStateOutput("%StateStoreName%", Key = "order")]
        public static JsonElement Run(
            [DaprServiceInvocationTrigger] JsonElement payload, 
            FunctionContext functionContext)
        {
            var log = functionContext.GetLogger("CreateNewOrder");
            log.LogInformation("C# function processed a CreateNewOrder request from the Dapr Runtime.");

            // payload must be of the format { "data": { "value": "some value" } }
            payload.TryGetProperty("data", out JsonElement data);

            return data;
        }
    }
}

File: samples/python-v2-azurefunctions/CreateNewOrder.py
# import logging
# import json
# import azure.functions as func


# def main(payload,
#          order: func.Out[str]) -> None:
#     logging.info(
#         'Python function processed a CreateNewOrder request from the Dapr Runtime.')
#     payload_json = json.loads(payload)
#     logging.info(payload_json["data"])
#     order.set(json.dumps({"value": payload_json["data"]}))


import azure.functions as func
import logging

app = func.FunctionApp()

@app.dapr_service_invocation_trigger(arg_name="dapr", method_name="CreateNewOrder") 
def dapr_service_trigger(json_payload: str):
    logging.info('Python EventHub trigger processed an event: %s', json_payload)